### DISCUSSION TITLE
DNS over tls fails: Context deadline exceeded

### DISCUSSION THREAD

**Discussion by Elekam** [qdm12/gluetun#2568] [Q&A]
Hello, Im experiencing that my vpn connection starts up, my torrent client starts downloading at full speed, and few minutes later halts to a near-stop, with the logs full of this line (with varying domains):
WARN [dns] dialing tls server for request IN A tracker.artixlinux.org.: context deadline exceeded

When I searched for this error in the discussions and issues, it mostly seems to be a configuration issue. But Im not sure what in my config is wrong? I seem to have filled everything out exactly as the wireguard config file from ProtonVPN says, public and private key are definitely correct. Would really appreciate any help here.

Attached is my docker compose and the logs after restarting the container and letting it run for a few minutes.
I am aware of the api warning about the new 3.40 update, but so far it seems its okay without it and I dont want to overcomplicate troubleshooting this. The api stuff comes after I get this issue fixed.

[gluetun_log.txt](https://github.com/user-attachments/files/17624245/gluetun_log.txt)
[docker-compose.txt](https://github.com/user-attachments/files/17624260/docker-compose.txt)

**Comment by epic0421** [comment]
It is likely related to [this issue](https://github.com/qdm12/gluetun/issues/2533). Try adding the WIREGUARD_MTU environment variable with a value of 1320 [(ref)](https://github.com/qdm12/gluetun/issues/2533#issuecomment-2453216691) and see if that fixes it.

**Reply by Elekam** [reply]
It did not, but thanks for making me aware of that issue :) Maybe it will work after that issue has been fixed, otherwise I need to dig deeper

**Reply by akshobhsingh** [reply]
Did you find a fix? Currently experiencing the same issue, even after lowering to WIREGUARD_MTU value.
